src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -717,6 +726,19 @@ protected virtual void OnProgressChanged(ProgressChangedEventArgs e)
                 handler(this, e);
         }
 
+        private void RunProgressDialog(IntPtr owner, object argument, CancellationToken token)
+        {
+            using (var cts = System.Threading.CancellationTokenSource.CreateLinkedTokenSource(token))
```

Comment:
`RunProgressDialog` starts the background worker **asynchronously**, so we unfortunately can't dispose the `CancellationTokenSource` at this point otherwise the background thread where the job is running will always get an invalid token.

We can only dispose the `CancellationTokenSource` after the job ends i.e. `RunWorkerCompleted`

Also, if this method is going to be the new entry point for starting the job, we need to move the validation here (check if job is already running)